libass/ass.c

```diff
@@ -1359,6 +1364,34 @@ void ass_flush_events(ASS_Track *track)
     track->parser_priv->read_order_elems = 0;
 }
 
+void ass_configure_gc(ASS_Track *track, int enabled, long long delay)
+{
+    track->parser_priv->gc_enabled = enabled;
+
+    if (0ll <= delay)
+        track->parser_priv->gc_delay = delay;
```

Comment:
This may behave surprisingly for a sequence like e.g. the following:

```C
ass_configure_gc(track, true, 5000);
ass_configure_gc(track, false, LONG_MAX);
ass_configure_gc(track, true, -1);
```

This will end up with a delay of `LONG_MAX` eventhough it wasn't requested during activation and not in use at any point in time before. Otoh the following options come to mind:
- ignore `delay` when `enabled == false` and explicitly document either the current behaviour for negative values or declare negative values to be illegal and subject to change
- refuse to change anything for illegal values and return `bool` to indicate whether configuration was successfull
- merge `enabled` and `delay` with negative values disabling gc

Reply:
remark obsoleted by merging `enabled` and `delay`